### PR TITLE
Download CMS validation file from FMS

### PR DESCRIPTION
**Pull request by omnivion**
### Information
- JIRA story QPPSF-5376.

### Changes proposed in this PR:
- Move cpc+ validation file from s3 to FMS

### Checklist 
- [ ] All JUnit tests pass (`mvn clean verify`).
- [ ] New unit tests written to cover new functionality.
- [ ] Added and updated JavaDocs for non-test classes and methods.
- [ ] No local design debt. Do you feel that something is "ugly" after your changes?
- [ ] Updated documentation (`README.md`, etc.) depending if the changes require it.

### CODE REVIEW

File: ERROR_MESSAGES.md [hunk 1]
```diff
@@ -66,7 +66,7 @@ Any text in the following format `(Example)` are considered variables to be fill
 * 72 : CT - The Performance Rate is missing
 * 78 : CT - The Program 'Mips Virtual Group' was found. The required entity id for this program name was missing. Please provide a virtual group identifier with the 'Mips Virtual Group' program name.
 * 79 : CT - There is no TIN validator present, so NPI/Alternative Payment Model (APM) combinations cannot be verified
-* 80 : CT - The given National Provider Identifier `(npi)`, Taxpayer Identification Number `(tin)`, and Alternative Payment Model `(apm)` are not a valid combination
```
Comment (saquino0827) [inline ERROR_MESSAGES.md]: This is a required change needed for formatting. The variables `(npi)` need to be set like this so they can be filtered and replaced. Please revert this change.

Comment (omnivion) [reply ERROR_MESSAGES.md]: I didn't change this, just rebuilt error messages file.

File: rest-api/src/main/java/gov/cms/qpp/conversion/api/model/Constants.java [hunk 2]
```diff
@@ -18,9 +18,9 @@
 	public static final String DYNAMO_CPC_ATTRIBUTE = "Cpc";
 	public static final String DYNAMO_CPC_PROCESSED_CREATE_DATE_ATTRIBUTE = "CpcProcessed_CreateDate";
 	public static final String DYNAMO_CREATE_DATE_ATTRIBUTE = "CreateDate";
-	public static final String CPC_PLUS_BUCKET_NAME_VARIABLE = "CPC_PLUS_BUCKET_NAME";
-	public static final String CPC_PLUS_FILENAME_VARIABLE = "CPC_PLUS_VALIDATION_FILE";
 	public static final String CPC_PLUS_UNPROCESSED_FILE_SEARCH_DATE_VARIABLE = "CPC_PLUS_UNPROCESSED_FILTER_START_DATE";
+	public static final String FMS_TOKEN = "";
+	public static final String AR_API_BASE_URL = "https://ar.qpp.internal/api";
```
Comment (saquino0827) [inline rest-api/src/main/java/gov/cms/qpp/conversion/api/model/Constants.java]: Does this URL not need to change based off environment?

File: rest-api/src/main/java/gov/cms/qpp/conversion/api/services/StorageServiceImpl.java [hunk 3]
```diff
@@ -107,19 +112,20 @@ public InputStream getFileByLocationId(String fileLocationId) {
...
-
-		GetObjectRequest getObjectRequest = new GetObjectRequest(bucketName, key);
-		S3Object s3Object = amazonS3.getObject(getObjectRequest);
-
-		return s3Object.getObjectContent();
+		API_LOG.info("Retrieving CPC+ validation file from FMS");
+
+		RestTemplate rest = new RestTemplate();
+		RequestEntity<?> entity = RequestEntity.get(
+					URI.create(Constants.AR_API_BASE_URL + "/v1/fms/file/qppct/cpc_plus_validations"))
```
Comment (saquino0827) [inline rest-api/src/main/java/gov/cms/qpp/conversion/api/services/StorageServiceImpl.java]: Does this retrieve all files in the `/v1/fms/file/qppct/cpc_plus_validations` directory in FMS?

File: ERROR_MESSAGES.md (hunk 1 above)
Comment (saquino0827) [reply ERROR_MESSAGES.md]: Ah my mistake.

File: rest-api/src/main/java/gov/cms/qpp/conversion/api/services/StorageServiceImpl.java (hunk 3 above)
Comment (omnivion) [reply rest-api/src/main/java/gov/cms/qpp/conversion/api/services/StorageServiceImpl.java]: Fixed for clarity

File: rest-api/src/main/java/gov/cms/qpp/conversion/api/model/Constants.java (hunk 2 above)
Comment (omnivion) [reply rest-api/src/main/java/gov/cms/qpp/conversion/api/model/Constants.java]: Fixed

File: rest-api/src/main/java/gov/cms/qpp/conversion/api/services/StorageServiceImpl.java [hunk 4]
```diff
@@ -107,19 +112,20 @@ public InputStream getFileByLocationId(String fileLocationId) {
 	 * @return file used for cpc+ validation.
 	 */
 	@Override
-	public InputStream getCpcPlusValidationFile() {
-		String bucketName = environment.getProperty(Constants.CPC_PLUS_BUCKET_NAME_VARIABLE);
-		String key = environment.getProperty(Constants.CPC_PLUS_FILENAME_VARIABLE);
-		if (StringUtils.isEmpty(bucketName) || StringUtils.isEmpty(key)) {
-			API_LOG.warn("No CPC+ bucket name and/or CPC+ key specified");
+	public byte[] getCpcPlusValidationFile() {
+		if (StringUtils.isEmpty(Constants.FMS_TOKEN)) {
```
Comment (saquino0827) [inline rest-api/src/main/java/gov/cms/qpp/conversion/api/services/StorageServiceImpl.java]: Will the FMS token be stored in S3 as an env variable? Because that constant will always be empty.

**Comment by saquino0827** [comment]
@omnivion I had one final comment about the `Constant.FMS_TOKEN`. Once that is addressed this PR LGTM

**Comment by saquino0827** [comment]
@omnivion Merging and beginning deployment/testing to move this along faster.
